lib/linter/linter.js

```diff
@@ -1001,7 +1003,11 @@ const internalSlotsMap = new WeakMap();
  */
 class Linter {
 
-    constructor() {
+    /**
+     * @param {Object} config the config object
+     * @param {string | undefined} config.cwd cwd, can be undefined.
+     */
+    constructor({ cwd }) {
```

Comment:
Should I make the `config` as `Object | undefined` for the sake of backward compatibility?

Reply:
Can we make this a non-breaking change by making the config object optional? e.g. `constructor({ cwd } = {}) {}`

Reply:
Yes, we should not make the breaking change.